catalog/dags/maintenance/rotate_envfiles.py

```diff
@@ -0,0 +1,320 @@
+                logger.info(
+                    "Found envfile %s for %s %s", envfile, environment, service_name
+                )
+                recent_envfiles[environment].add(envfile)
+
+    return recent_envfiles
+
+
+@task
+def identify_stale_envfiles(
+    lt_envfiles_to_keep: dict[str, set[str]],
+    task_def_envfiles_to_keep: dict[str, set[str]],
+) -> dict[str, set[str]]:
+    """
+    Retrieve the envfiles in the environment's environment file bucket
+    that are NOT included in the ``envfiles_to_keep`` list.
+    """
+    s3_hook = S3Hook(aws_conn_id=_get_conn_id())
+
+    stale_envfiles = defaultdict(set)
+    for env in ("staging", "production"):
+        bucket = f"openverse-{env}-environment-files"
+        envfiles_to_keep = lt_envfiles_to_keep[env] | task_def_envfiles_to_keep[env]
+        objects = s3_hook.conn.list_objects_v2(Bucket=bucket)
+
+        for obj in objects["Contents"]:
+            if obj["Key"] not in envfiles_to_keep:
+                stale_envfiles[env].add(obj["Key"])
+
+        logger.info(
+            "Found %s stale %s envfiles: %s",
+            len(stale_envfiles[env]),
+            env,
+            ", ".join(stale_envfiles[env]),
+        )
+
+    return stale_envfiles
+
+
+@task
+def delete_stale_envfiles(
+    envfiles_to_delete: dict[str, set[str]],
+) -> dict[str, list[str]]:
+    s3_hook = S3Hook(aws_conn_id=_get_conn_id())
+
+    is_deletion_enabled = Variable.get(
+        "ENABLE_S3_ENVFILE_DELETION", False, deserialize_json=True
+    )
+
+    deleted_objects: dict[str, list[str]] = {}
+    for env, envfiles in envfiles_to_delete.items():
+        if not envfiles:
+            logger.info("No envfiles to delete for %s", env)
+            continue
+
+        bucket = f"openverse-{env}-environment-files"
+        delete_request = {"Objects": [{"Key": envfile} for envfile in envfiles]}
+        if not is_deletion_enabled:
+            logger.info(
+                "Envfile deletion disabled. The following files would have been deleted from %s: %s",
+                bucket,
+                ", ".join(envfiles),
+            )
+            continue
+
+        delete_response = s3_hook.conn.delete_objects(
+            Bucket=bucket,
+            Delete=delete_request,
+        )
+        if errors := delete_response.get("Errors", []):
+            logger.error(
+                "Received the following errors when issuing the delete request: %s",
+                "; ".join(
+                    [
+                        f"{error['Key']} {error['Code']}: {error['Message']}"
+                        for error in errors
+                    ]
+                ),
+            )
+
+        if deleted := delete_response.get("Deleted", []):
+            deleted_objects[env] = [obj["Key"] for obj in deleted]
+            logger.info(
+                "Deleted the following objects: %s", "; ".join(deleted_objects[env])
+            )
```

Comment:
Same as previous lists.

```suggestion
            logger.info(
                "Deleted the following objects: %s", ";\n\t".join(deleted_objects[env])
            )
```